library/model.c

```diff
@@ -14,14 +14,12 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+#include <time.h>
 
 #if _WIN32
 #define timegm(v) _mkgmtime(v)
 #else
 #define _GNU_SOURCE
```

Comment:
If I remember correctly, _GNU_SOURCE has to be defined before #include <time.h> otherwise it has no effect

Reply:
i've added the include in both branches of the if to address this. i chose to do that instead of adding another if block. figured it was marginally better than another if...